providers/implementations/exchange/dh_exch.c

```diff
@@ -293,6 +293,8 @@ static void *dh_dupctx(void *vpdhctx)
             goto err;
     }
     dstctx->kdf_cekalg = OPENSSL_strdup(srcctx->kdf_cekalg);
+    if (dstctx->kdf_cekalg == NULL)
```

Comment:
I think it is possible for `srcctx->kdf_cekalg` to be NULL, in which case it would not be an error for `dtsctx->kdf_cekalg` to also be NULL.

Reply:
ok, I will close it.

Reply:
Well, I didn't say there wasn't a problem to be fixed. In the event that `srcctx->kdf_cekalg` is NULL then it is valid for `dstctx->kdf_cekalg` to be NULL. However if `srcctx->kdf_cekalg` is non-NULL then `dstctx->kdf_cekalg` should also be non-NULL. So the check needs to be a bit more clever than how you currently wrote it - but a check is still needed, e.g.
````
if (srctx->kdf_cekalg != NULL) {
    dsctx->kdf_cekalg = OPENSSL_strdup(srcctx->kdf_cekalg);
    if (dstctx->kdf_cekalg == NULL)
        goto err;
}
````

Reply:
Thanks, I have sent a fixup.